src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, ElementRef, Inject, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { faCircle, faExpand, faPaperPlane, faXmark } from '@fortawesome/free-solid-svg-icons';
+import { MAT_DIALOG_DATA, MatDialog } from '@angular/material/dialog';
+import { IrisStateStore } from 'app/iris/state-store.service';
+import { ConversationErrorOccurredAction, StudentMessageSentAction } from 'app/iris/message-store.model';
+import { IrisHttpMessageService } from 'app/iris/http-message.service';
+import { IrisClientMessage, IrisMessage, IrisSender } from 'app/entities/iris/iris-message.model';
+import { IrisMessageContent, IrisMessageContentType } from 'app/entities/iris/iris-content-type.model';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'jhi-exercise-chat-widget',
+    templateUrl: './exercise-chat-widget.component.html',
+    styleUrls: ['./exercise-chat-widget.component.scss'],
+})
+export class ExerciseChatWidgetComponent implements OnInit, OnDestroy {
+    @ViewChild('chatWidget') chatWidget!: ElementRef;
+    @ViewChild('chatBody') chatBody!: ElementRef;
+
+    readonly SENDER_USER = IrisSender.USER;
+    readonly SENDER_SERVER = IrisSender.LLM;
+    readonly stateStore: IrisStateStore;
+
+    stateSubscription: Subscription;
+    messages: IrisMessage[] = [];
+    newMessageTextContent = '';
+    isLoading: boolean;
+    sessionId: number;
+    error = ''; // TODO: error object
+    dots = 1;
+
+    constructor(private dialog: MatDialog, @Inject(MAT_DIALOG_DATA) public data: any, private httpMessageService: IrisHttpMessageService) {
+        this.stateStore = data.stateStore;
+    }
+
+    // Icons
+    faPaperPlane = faPaperPlane;
+    faCircle = faCircle;
+    faExpand = faExpand;
+    faXmark = faXmark;
+
+    ngOnInit() {
+        this.scrollToBottom('auto');
+        this.animateDots();
+        this.stateSubscription = this.stateStore.getState().subscribe((state) => {
+            this.messages = state.messages as IrisMessage[];
+            this.isLoading = state.isLoading;
+            this.error = state.error;
+            this.sessionId = Number(state.sessionId);
+        });
+    }
+
+    ngOnDestroy() {
+        this.stateSubscription.unsubscribe();
+    }
+
+    animateDots() {
+        setInterval(() => {
+            this.dots = this.dots < 3 ? (this.dots += 1) : (this.dots = 1);
+        }, 500);
+    }
+
+    onSend(): void {
+        if (this.newMessageTextContent) {
+            const message = this.newUserMessage(this.newMessageTextContent);
+            this.stateStore
+                .dispatchAndThen(new StudentMessageSentAction(message))
+                .then(() => this.httpMessageService.createMessage(<number>this.sessionId, message).toPromise())
+                .then(() => this.scrollToBottom())
+                .catch(() => this.stateStore.dispatch(new ConversationErrorOccurredAction('Something went wrong. Please try again later!')));
```

Comment:
@Hialus because nobody is using our feature currently, could we leave the translation part in the next PR for this task https://jira.ase.in.tum.de/browse/PCIL23UX-35?